federation/executor.go

```diff
@@ -39,65 +43,41 @@ func fetchSchema(ctx context.Context, e ExecutorClient, optionalArgs interface{}
 }
 
 func NewExecutor(ctx context.Context, executors map[string]ExecutorClient) (*Executor, error) {
-	return NewExecutorWithOptionalArgs(ctx, executors, nil)
+	return NewCustomExecutor(ctx, executors, nil, nil, 5)
 }
 
-func NewExecutorWithOptionalArgs(ctx context.Context, executors map[string]ExecutorClient, optionalArgs interface{}) (*Executor, error) {
-	// Fetches the schemas from the executors clients
-	schemas := make(map[string]*introspectionQueryResult)
-	for server, client := range executors {
-		schema, err := fetchSchema(ctx, client, optionalArgs)
-		if err != nil {
-			return nil, oops.Wrapf(err, "fetching schema %s", server)
-		}
-
-		var iq introspectionQueryResult
-		if err := json.Unmarshal(schema, &iq); err != nil {
-			return nil, oops.Wrapf(err, "unmarshaling schema %s", server)
-		}
-
-		schemas[server] = &iq
+func NewCustomExecutor(ctx context.Context, executors map[string]ExecutorClient, schemaSyncer SchemaSyncer, optionalArgs interface{}, schemaSyncInterval int64) (*Executor, error) {
```

Comment:
What is the unit of the interval, maybe rename it to `schemaSyncInternvalSecond`